### PR TITLE
allow setting of NODE_ENV

### LINKED ISSUE #4: Debug mode
Support `NODE_ENV` set to `development` to enable debug logging in the CLI and in the Worker itself (if not done with a `--debug` argument or similar).

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
resolves #4

### CODE REVIEW

File: src/buildApplication/buildGlobalContext.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * @returns the plain javascript string that should be added at the top of the the _worker.js file
+ */
+export function generateGlobalJs(): string {
+	const nodeEnv = getNodeEnv();
+	return `globalThis.process = { env: { NODE_ENV: '${nodeEnv}' } };`;
+}
+
+function getNodeEnv(): string {
+	const nextJsNodeEnvs = ['production', 'development', 'test'];
+	const defaultNodeEnv = nextJsNodeEnvs[0];
```
Comment (GregBrimble) [inline src/buildApplication/buildGlobalContext.ts:16]: Nit: to rely on the ordering of this array is pretty brittle, imo. I'd personally be explicit, or better yet, use an enum.

```suggestion
	const defaultNodeEnv = 'production';
```

**Review by GregBrimble — APPROVED** [review]

File: src/buildApplication/buildGlobalContext.ts (hunk 1 above)
Comment (dario-piotrowicz) [reply src/buildApplication/buildGlobalContext.ts]: I've added a unit test which makes sure that the default value is `production` does that help? 🙂

Comment (dario-piotrowicz) [reply src/buildApplication/buildGlobalContext.ts]: I've added a string enum anyways 🙂👍

**Review by GregBrimble — APPROVED** [review]
✅✅

**Pull request merged**